app/backend/src/couchers/jobs/handlers.py

```diff
@@ -40,3 +44,74 @@ def process_purge_signup_tokens(payload):
     logger.info(f"Purging signup tokens")
     with session_scope() as session:
         session.query(SignupToken).filter(SignupToken.is_valid == False).delete(synchronize_session=False)
+
+
+def process_send_message_notifications(payload):
+    """
+    Sends out email notifications for messages that have been unseen for a long enough time
+    """
+    # very crude and dumb algorithm
+    logger.info(f"Sending out email notifications for unseen messages")
+
+    with session_scope() as session:
+        # users who have unnotified messages older than 30 minutes in any group chat
+        users = (
+            session.query(User)
+            .join(GroupChatSubscription, GroupChatSubscription.user_id == User.id)
+            .join(Message, Message.conversation_id == GroupChatSubscription.group_chat_id)
+            .filter(Message.time >= GroupChatSubscription.joined)
+            .filter(or_(Message.time <= GroupChatSubscription.left, GroupChatSubscription.left == None))
+            .filter(Message.id > User.last_notified_message_id)
+            .filter(Message.id > GroupChatSubscription.last_seen_message_id)
+            .filter(Message.time < now() - timedelta(minutes=30))
+            .filter(Message.message_type == MessageType.text)  # TODO: only text messages for now
+            .group_by(User)
```

Comment:
Is the group_by needed if you are only asking for a list of users?

Reply:
Probably not, seems to be leftover from trying other things.